web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  ChangeEvent,
+  MouseEvent,
+  Suspense,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import { SubmitHandler, useWatch } from 'react-hook-form';
+import { CreditsAmount } from 'web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount';
+import {
+  cryptoOptions,
+  DEFAULT_CRYPTO_CURRENCY,
+} from 'web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount.constants';
+import Form from 'web-marketplace/src/components/molecules/Form/Form';
+import { useZodForm } from 'web-marketplace/src/components/molecules/Form/hook/useZodForm';
+
+import Card from 'web-components/src/components/cards/Card';
+import {
+  CURRENCIES,
+  Currency,
+} from 'web-components/src/components/DenomIconWithCurrency/DenomIconWithCurrency.constants';
+import { Loading } from 'web-components/src/components/loading';
+
+import { AdvanceSettings } from './ChooseCreditsForm.AdvanceSettings';
+import { PAYMENT_OPTIONS } from './ChooseCreditsForm.constants';
+import { CryptoOptions } from './ChooseCreditsForm.CryptoOptions';
+import { PaymentOptions } from './ChooseCreditsForm.PaymentOptions';
+import {
+  chooseCreditsFormSchema,
+  ChooseCreditsFormSchemaType,
+} from './ChooseCreditsForm.schema';
+import { PaymentOptionsType } from './ChooseCreditsForm.types';
+
+export function ChooseCreditsForm({
+  creditVintages,
+  creditsAvailable = [
+    {
+      credits: 2000,
+      currency: 'usd',
+    },
+  ],
+}: {
+  creditVintages: { date: string; credits: string; batchDenom: string }[];
+  creditsAvailable: {
+    credits: number;
+    currency: Currency;
+  }[];
+}) {
+  const [paymentOption, setPaymentOption] = useState<PaymentOptionsType>(
+    PAYMENT_OPTIONS.CARD,
+  );
+  const [advanceSettingsOpen, setAdvanceSettingsOpen] = useState(false);
+
+  const toggleAdvancedSettings = (e: MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    setAdvanceSettingsOpen(prev => !prev);
+  };
+
+  const [currency, setCurrency] = useState<Currency>(
+    paymentOption === PAYMENT_OPTIONS.CRYPTO
+      ? DEFAULT_CRYPTO_CURRENCY
+      : CURRENCIES.usd,
+  );
+
+  const form = useZodForm({
+    schema: chooseCreditsFormSchema,
+    defaultValues: {
+      amountCurrency: 1,
+      amountCredits: 1,
+      cryptoPurchaseOption: cryptoOptions?.[0]?.label,
```

Comment:
instead of using the displayed label as value here, we could probably just use a boolean: `retiring` like I did in https://github.com/regen-network/regen-web/pull/2419